### DISCUSSION TITLE
Cannot install yolox

### DISCUSSION THREAD

**Discussion by Jalilnkh** [Megvii-BaseDetection/YOLOX#1667] [Q&A]
I am trying to install Yolox using the following command:

pip install yolox
I got this command from the following link:

https://pypi.org/project/yolox/

I got the following error:

Preparing metadata (setup.py) ... error

 error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.

  │ exit code: 1

  ╰─> [8 lines of output]

  Traceback (most recent call last):

    File "<string>", line 2, in <module>

    File "<pip-setuptools-caller>", line 34, in <module>

    File "/tmp/pip-install-e8uz9eo5/yolox_dd607ece87a3411aabef0d9d5bff591a/setup.py", line 77, in <module>

      install_requires=get_install_requirements(),

    File "/tmp/pip-install-e8uz9eo5/yolox_dd607ece87a3411aabef0d9d5bff591a/setup.py", line 49, in get_install_requirements

      with open("requirements.txt", "r", encoding="utf-8") as f:

  FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'

  [end of output]

 note: This error originates from a subprocess, and is likely not a problem with pip.
 error: metadata-generation-failed

  × Encountered error while generating package metadata.
  ╰─> See above for output.

 note: This is an issue with the package mentioned above, not pip.
 hint: See above for details.
How can I solve this problem? I should mention that I am installing this package on a remote machine based on ubuntu.

**Comment by FateScript** [comment]
Hi @Jalilnkh , It seems that you are installing YOLOX through pypi, did you try the fixed version, e.g. `pip install yolox==0.3.0`,
If pypi still not working, try install yolox from source, it should help.

**Comment by kd10041** [comment]
Hey @FateScript I am encountering the same error I tried installing for a fixed version and also tried installing from the github repository. Can you please help regarding the error.
I am facing error like following for all 3 installation process : 
DEPRECATION: Loading egg at c:\users\k.divyanshu\appdata\local\programs\python\python312\lib\site-packages\yolox-0.3.0-py3.12.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [32 lines of output]
      fatal: not a git repository (or any of the parent directories): .git
      fatal: not a git repository (or any of the parent directories): .git

**Comment by WASasquatch** [comment]
I have the same problem. And if I install 0.3.0, I cannot import yolox

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], [line 9](vscode-notebook-cell:?execution_count=1&line=9)
      [7](vscode-notebook-cell:?execution_count=1&line=7) import json
      [8](vscode-notebook-cell:?execution_count=1&line=8) from ultralytics import YOLO
----> [9](vscode-notebook-cell:?execution_count=1&line=9) from yolox.tracker.byte_tracker import BYTETracker, STrack  # Correct BYTETracker import from YOLOX
     [11](vscode-notebook-cell:?execution_count=1&line=11) # Function to perform OCR using OpenAI's GPT-4o-mini
     [12](vscode-notebook-cell:?execution_count=1&line=12) def perform_ocr(client, pil_image):
     [13](vscode-notebook-cell:?execution_count=1&line=13)     # Convert PIL image to bytes

ModuleNotFoundError: No module named 'yolox'
```

**Comment by maxicuevas** [comment]
To my knowledge, YOLOX version 0.3.0 can be installed in python version 3.8 using pip version 22.1.2, but it fails with newer pip versions.
Are you planning to release a PyPI package of YOLOX 0.3.0 for newer pip versions?
